### PR TITLE
API Authorization Audit

### PR DESCRIPTION
**Pull request by Senjai**
This had to be done to remove the explicit has_spree_role? checks we had littered through the API. 

Done to ensure conventions established in #136 hold true for the API as well.

Most of this should be pretty straight forward, given that having the admin role is a blanket manage everything permission, we explicitly check for the admin permission on records of interest. 

This follows the convention in https://github.com/solidusio/solidus/blob/master/backend/app/controllers/spree/admin/base_controller.rb#L22-L23 with the expectation that authorization is also performed for the action (e.g. `:create`, or `:update`)

Also includes minor cleanup.

### CODE REVIEW

File: api/app/controllers/spree/api/checkouts_controller.rb [hunk 1]
```diff
@@ -52,7 +52,7 @@ def update
         authorize! :update, @order, order_token
 
         if @order.update_from_params(params, permitted_checkout_attributes, request.headers.env)
-          if current_api_user.has_spree_role?('admin') && user_id.present?
+          if can?(:admin, @order) && user_id.present?
```
Comment (athal7) [inline api/app/controllers/spree/api/checkouts_controller.rb]: just checking - does the owner of an order have the ability to admin an order? if so, could they change the user_id? that would be bad

File: api/app/controllers/spree/api/orders_controller.rb [hunk 2]
```diff
@@ -30,13 +27,14 @@ def cancel
 
       def create
         authorize! :create, Order
-        order_user = if @current_user_roles.include?('admin') && order_params[:user_id]
+
+        order_user = if order_params[:user_id]
```
Comment (athal7) [inline api/app/controllers/spree/api/orders_controller.rb]: same concern here as above about creating for an alternative user

File: api/app/controllers/spree/api/orders_controller.rb [hunk 3]
```diff
@@ -66,17 +64,9 @@ def show
...
-        else
-          # This will reset checkout back to address and delete all shipments.
-          @order.contents.update_cart(order_params)
-        end
-
-        if result
+        if @order.contents.update_cart(order_params)
           user_id = params[:order][:user_id]
-          if current_api_user.has_spree_role?('admin') && user_id
+          if can?(:admin, @order) && user_id
```
Comment (athal7) [inline api/app/controllers/spree/api/orders_controller.rb]: and here as well

File: api/app/controllers/spree/api/orders_controller.rb [hunk 4]
```diff
@@ -30,13 +27,14 @@ def cancel
...
-        order_user = if @current_user_roles.include?('admin') && order_params[:user_id]
+
+        order_user = if order_params[:user_id]
           Spree.user_class.find(order_params[:user_id])
         else
           current_api_user
         end
 
-        import_params = if @current_user_roles.include?("admin")
+        import_params = if can?(:admin, Spree::Order)
```
Comment (athal7) [inline api/app/controllers/spree/api/orders_controller.rb]: and here :)

File: api/app/controllers/spree/api/checkouts_controller.rb (hunk 1 above)
Comment (athal7) [inline api/app/controllers/spree/api/checkouts_controller.rb]: from offline conversation it sounds like that is not the case. i have some concern about a user acquiring this role either in base solidus or in a store at some point. is it worth adding an extra warning or spec somewhere around that?

**Comment by athal7** [comment]
lgtm

File: api/app/controllers/spree/api/checkouts_controller.rb (hunk 1 above)
Comment (jhawthorn) [inline api/app/controllers/spree/api/checkouts_controller.rb]: I think this is the reason for the `:admin` pseudo-action. All users can edit/update their own order, so admin is to distinguish between that kind of editing and this kind. A good clarification of this somewhere would of course be welcome.

**Comment by jhawthorn** [comment]
:+1:

**Comment by magnusvk** [comment]
@athal7 I assume your 'lgtm' constitutes the second thumb here? Can I hit merge on this? Or feel free to do so yourself.

**Pull request merged**